### PR TITLE
refactor: extract tx broadcast from component and add unit tests

### PR DESCRIPTION
**Pull request by lucachaco**
Extract tx broadcast code from component and add unit test (pending)

**The following cases will be in a separate PR:**

Segwit RBTC to Segwit RBTC (Partial funds))

USDT to USDT (Partial funds))

Legacy BTC to Legacy BTC (All funds)

USDT to USDT (All funds)
## Motivation and Context
Decouple logic from component to make it easier to test

## How Has This Been Tested?
Yes, locally

## Types of changes

- [ ] fix: bug fixes, e.g. fix Button color on DarkTheme.
- [ ] feat: new features, e.g. add Snackbar component.
- [x] refactor: code refactor, e.g. new folder structure for components.
- [ ] docs: changes into documentation, e.g. add usage example for Button.
- [ ] test: adding or updating tests, eg unit, snapshot testing.
- [ ] chore: tooling changes, e.g. change circleci config.
- [ ] BREAKING CHANGE: for changes that break existing usage, e.g. change API of a component.

## Checklist:

- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)

* **Other information**:

* **SCREENSHOTS**:

### CODE REVIEW

**Comment by lucachaco** [comment]
Screen Shot 2021-05-18 at 22 34 47

**Review by patogallaiovlabs — CHANGES_REQUESTED** [review]

File: src/services/transactionServices.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}) => {
+  const {
+    symbol, balance, precision,
+  } = coin;
+  const extraParams = { data: '', memo, gasFee: feeParams };
+  let finalToAddress = toAddress;
+  // In order to send all all balances, we cannot use the amount in the text box to calculate the amount sent, but use the coin balance.
+  // The amount of the text box is fixed decimal places
+  let value = new BigNumber(amount);
+  if (isRequestSendAll) {
```
[the remark this answers is not in the document]
Comment (lucachaco) [reply src/services/transactionServices.js:16]: Yes, good point. As discussed we cant do it in a different ticket.

File: __test__/transactionServices.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    });
+
+
+    it('should build a transaction with a RBTC address when sending all', async () => {
+        const builtTransaction = await buildTransaction(segwitRbtcSendAllInput);
+        expect(builtTransaction).toHaveProperty('receiver');
+        expect(builtTransaction.receiver).toBe(segwitRbtcSendAllInput.toAddress.toLowerCase());
+        expect(builtTransaction).toHaveProperty('sender');
+        expect(builtTransaction.sender).toBe(segwitRbtcSendAllInput.coin.address);
+        expect(builtTransaction.symbol).toBe(segwitRbtcSendAllInput.coin.symbol);
```
[the remark this answers is not in the document]
Comment (lucachaco) [reply __test__/transactionServices.test.js:83]: Added.

File: src/pages/wallet/transfer.js [hunk 3]
```diff
@@ -914,38 +915,24 @@ class Transfer extends Component {
...
+      });
+
       this.setState({ loading: false });
       const completedParams = {
         coin,
         hash: transaction.txHash,
       };
       navigation.navigate('TransferCompleted', completedParams);
-      transaction = null;
+      // transaction = null;
```
[the remark this answers is not in the document]
Comment (lucachaco) [reply src/pages/wallet/transfer.js]: Removed..

**Review by patogallaiovlabs — APPROVED** [review]

**Pull request merged**